bionic/core/flow_execution.py

```diff
@@ -108,32 +92,59 @@ def run(self, states):
             if self._parallel_execution_enabled:
                 self._bootstrap.executor.stop_logging()
 
-    def _process_entry(self, entry):
+    def _process_active_entry(self, entry):
+        """
+        Takes the current active entry and does the minimal amount of work to move it
+        to the next stage. When this function returns, the entry will no longer be in
+        the ACTIVE stage: it will be COMPLETED, IN_PROGRESS, or BLOCKED, and we will
+        be able to move on to the next entry.
+        """
+
         assert entry.stage == EntryStage.ACTIVE
 
-        state = entry.state
-        # Initialize the task state before attempting to complete it.
-        state.initialize(self._bootstrap, self._flow_instance_uuid)
+        # In theory we could have an entry that doesn't require priming, but in practice
+        # we always require it when we create one.
+        assert entry.is_requested_to_be_primed
 
-        # Attempt to complete the task state from persistence cache.
-        state.attempt_to_complete_from_cache()
-        if state.is_cached:
+        # First, if we already have a cached value, we can stop immediately without
+        # needing to process any dependencies.
+        entry.state.set_up_caching_flags(self._bootstrap)
```

Comment:
I don't think you need to set up the caching flags before initialize anymore. I had to set them up earlier because we used the caching flags to determine whether dependency entries were needed in memory.